src/charm.py

```diff
@@ -582,26 +597,98 @@ def _wp_is_installed(self):
         logger.debug("Check if WordPress is installed")
         return self._run_wp_cli(["wp", "core", "is-installed"]).return_code == 0
 
-    def _current_effective_db_info(self):
+    @property
+    def _config_database_config(self) -> types_.DatabaseConfig | None:
+        """Database configuration details from charm config.
+
+        Returns:
+            Database configuration required to establish database connection.
+            None if not exists.
+        """
+        if any(
```

Comment:
This would be `True` if only one of the items in the list is actually present, is that as intended?

Reply:
Yes, if any of the keys are provided, it assumes that the charm will use database connection values from configuration.